docs/modelsvalidator/validate.py

```diff
@@ -0,0 +1,382 @@
+                )
+        if type == "number":
+            valid_attributes.append("minimum")
+            if not isinstance(field.get("minimum", 0), int):
+                self.errors.append(f"'minimum' for {collectionfield} is not a number.")
+        if type == "string":
+            valid_attributes.append("maxLength")
+            if not isinstance(field.get("maxLength", 0), int):
+                self.errors.append(
+                    f"'maxLength' for {collectionfield} is not a number."
+                )
+        if type in DATA_TYPES:
+            valid_attributes.append(
+                "default"
+            )  # TODO: this is not checked, if the datatype is correct.
+            if "default" in field:
+                self.validate_value_for_type(type, field["default"], collectionfield)
+            valid_attributes.append("enum")
+            if "enum" in field:
+                if not isinstance(field["enum"], list):
+                    self.errors.append(f"'enum' for {collectionfield}' is not a list.")
+                for value in field["enum"]:
+                    self.validate_value_for_type(type, value, collectionfield)
+
+        if type in RELATION_TYPES:
+            valid_attributes.append("on_delete")
+            if "on_delete" in field and field["on_delete"] not in (
+                "CASCADE",
+                "PROTECT",
+            ):
+                self.errors.append(
+                    f"invalid value for 'on_delete' for {collectionfield}"
+                )
+            valid_attributes.append("equal_fields")
+
+        if type == "template":
+            if "$" not in field_name:
+                self.errors.append(
+                    f"The template field {collectionfield} is missing a $"
+                )
+            valid_attributes.append("replacement_collection")
+        elif "$" in field_name and not nested:
+            print(field_name, field)
+            self.errors.append(f"The non-template field {collectionfield} contains a $")
+
+        for attr in field.keys():
+            if attr not in valid_attributes:
+                self.errors.append(
+                    f"Attribute '{attr}' for collectionfield {collectionfield} is invalid."
+                )
+
+        if not isinstance(field.get("description", ""), str):
+            self.errors.append(f"Description of {collectionfield} must be a string.")
+
+        if type == "template":
+            self.check_field(collection, field_name, field["fields"], nested=True)
+
+    def validate_value_for_type(
+        self, type: str, value: Any, collectionfield: str
+    ) -> None:
+        basic_types = {
+            "string": str,
+            "number": int,
+            "boolean": bool,
+            "HTMLStrict": str,
+            "HTMLPermissive": str,
+            "timestamp": int,
+        }
+        if type in basic_types:
+            if not isinstance(value, basic_types[type]):
```

Comment:
This does not work for integers, I had an issue in the datavalidator previously: `bool` inherits from `int` and therefore `True`/`False` are valid int values which is not wanted.

Either handle this case separately or check the type with `type(...) == xy` which should yield the same results for the other types, since we are just dealing with basic types here.